roles/k8s-gpu-plugin/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: install pip
+  apt:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - "python-pip"
+  when: ansible_distribution == 'Ubuntu'
+
+- name: install openshift python client for k8s_raw module
+  pip:
+    name: openshift
+  when: ansible_distribution == 'Ubuntu'
+
+- name: install openshift python client for k8s_raw module
+  yum:
+    name: python2-openshift
+  when: ansible_os_family == 'RedHat'
+
+- name: create GPU device plugin
```

Comment:
Running with Ubuntu on dvt03, I'm seeing consistent failures at this step. Weirdly, it's consistently complaining about the openshift python client not being installed, despite the pip install two tasks earlier succeeding, and manual checks show that it's installed.

```
TASK [k8s-gpu-plugin : create GPU device plugin] *********************************************************************task path: /home/ubuntu/deepops/roles/k8s-gpu-plugin/tasks/main.yml:20
fatal: [dvt03]: FAILED! => {"changed": false, "msg": "This module requires the OpenShift Python client. Try `pip install openshift`"}
```

When I run in highly-verbose mode (`-vvv`) it shows a more interesting error indicating a dependency issue with the python requests module.

```
<dvt03> Failed to connect to the host via ssh: /usr/lib/python2.7/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.24.1) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
/usr/lib/python2.7/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.24.1) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
```

I'm going to re-provision again and see if this is reproducible. @dholt , did you see any similar issues when you tested?

Reply:
I saw this too, it's only when you run the new `nvidia-dgx` role, then the `k8s-cluster` playbook that it shows up.

Reply:
Looks like we probably just need to remove the older `requests` and `urllib3` packages from the Ubuntu repos and let the newer versions get installed via Pip:

Just running `k8s-cluster` on vanilla Ubuntu:

```
ii  python3-requests                      2.18.4-2ubuntu0.1                           all          elegant and simple HTTP library for Python3, built for human beings
ii  python3-requests-unixsocket           0.1.5-3                                     all          Use requests to talk HTTP via a UNIX domain socket - Python 3.x
ii  python3-urllib3                       1.22-1                                      all          HTTP library with thread-safe connection pooling for Python3
```

Just running `nvidia-dgx` on vanilla Ubuntu:

```
ii  python-requests                       2.18.4-2ubuntu0.1                           all          elegant and simple HTTP library for Python2, built for human beings
ii  python3-requests                      2.18.4-2ubuntu0.1                           all          elegant and simple HTTP library for Python3, built for human beings
ii  python3-requests-unixsocket           0.1.5-3                                     all          Use requests to talk HTTP via a UNIX domain socket - Python 3.x
ii  python-urllib3                        1.22-1                                      all          HTTP library with thread-safe connection pooling for Python
ii  python3-urllib3                       1.22-1                                      all          HTTP library with thread-safe connection pooling for Python3
```